astroquery/cadc/cadctap/job.py

```diff
@@ -0,0 +1,226 @@
+        else:
+            # async: result is in the server once the job is finished
+            self.__load_async_job_results()
+            return self.results
+
+    def save_results(self, filename, verbose=False):
+        """Saves job results
+        If the job is asynchronous, this method will block until the results
+        are available.
+
+        Parameters
+        ----------
+        filename : str, mandatory
+            name of the file to save the output to
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+        Reason for change
+        -----------------
+        Add redirects if resutls are in a different place and get right
+        output format for write()
+        """
+        output = filename
+        output_format = self.parameters['format']
+        if self._Job__resultInMemory:
+            if output_format == 'csv':
+                output_format = 'ascii.csv'
+            if "tsv" == output_format:
+                output_format = "ascii.fast_tab"
+
+            self.results.write(output, format=output_format)
+        else:
+            if not self.async_:
+                # sync: cannot access server again
+                print("No results to save")
+            else:
+                # Async
+                self.wait_for_job_end(verbose)
+                context = 'async/' + str(self.jobid) + "/results/result"
+                response = self.connHandler.execute_get(context)
+                if verbose:
+                    print(response.status, response.reason)
+                    print(response.getheaders())
+
+                response = self.redirectJob(response, verbose)
+                self.connHandler.dump_to_file(output, response)
+
+    def wait_for_job_end(self, verbose=False):
+        """Waits until a job is finished
+
+        Parameters
+        ----------
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+        Reason for change
+        -----------------
+        Wait less for first few tries then wait longer and
+        send abort if there is a keyboard interrupt
+        """
+        currentResponse = None
+        responseData = None
+        loops = 0
+        while True:
+            try:
+                responseData = self.get_phase(update=True)
+                currentResponse = self._Job__last_phase_response_status
+
+                lphase = responseData.lower().strip()
+                if verbose:
+                    print("Job " + self.jobid + " status: " + lphase)
+                if "pending" != lphase and "queued" != lphase and \
+                        "executing" != lphase:
+                    break
+                # PENDING, QUEUED, EXECUTING, COMPLETED, ERROR, ABORTED,
+                # UNKNOWN, HELD, SUSPENDED, ARCHIVED:
+                if loops < 10:
+                    seconds = 1
+                else:
+                    seconds = 30
```

Comment:
consider making these configurable.  But, no big deal if they're hard-coded for now

Reply:
Rather not for the reasons I've explained earlier.